nf_core/pipeline-template/lib/WorkflowMain.groovy

```diff
@@ -2,54 +2,8 @@
 // This file holds several functions specific to the main.nf workflow in the {{ name }} pipeline
 //
 
-import nextflow.Nextflow
-
 class WorkflowMain {
 
-    //
-    // Citation string for pipeline
-    //
-    public static String citation(workflow) {
-        return "If you use ${workflow.manifest.name} for your analysis please cite:\n\n" +
-            // TODO nf-core: Add Zenodo DOI for pipeline after first release
-            //"* The pipeline\n" +
-            //"  https://doi.org/10.5281/zenodo.XXXXXXX\n\n" +
-            "* The nf-core framework\n" +
-            "  https://doi.org/10.1038/s41587-020-0439-x\n\n" +
-            "* Software dependencies\n" +
-            "  https://github.com/${workflow.manifest.name}/blob/master/CITATIONS.md"
-    }
-
-
-    //
-    // Validate parameters and print summary to screen
-    //
-    public static void initialise(workflow, params, log) {
-
-        // Print workflow version and exit on --version
-        if (params.version) {
-            String workflow_version = NfcoreTemplate.version(workflow)
-            log.info "${workflow.manifest.name} ${workflow_version}"
-            System.exit(0)
-        }
-
-        // Check that a -profile or Nextflow config has been provided to run the pipeline
-        NfcoreTemplate.checkConfigProvided(workflow, log)
-
-        // Check that conda channels are set-up correctly
-        if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
-            Utils.checkCondaChannels(log)
-        }
-
-        // Check AWS batch settings
-        NfcoreTemplate.awsBatch(workflow, params)
-
-        // Check input has been provided
-        if (!params.input) {
-            Nextflow.error("Please provide an input samplesheet to the pipeline e.g. '--input samplesheet.csv'")
-        }
-    }
-
     {%- if igenomes %}
```

Comment:
what about the igenomes bit?

Reply:
https://github.com/nf-core/modules/issues/3864

Reply:
Since igenomes is optional in case you are creating a customised template, I think the best is to keep it in this groovy function